static/src/js/util/accessMethods/buildAccessMethods.js

```diff
@@ -25,202 +19,61 @@ export const buildAccessMethods = (collectionMetadata, isOpenSearch) => {
-          supportsVariableSubsetting: supportsVariableSubsetting(serviceItem),
-          type: serviceType,
-          variables
-        }
-      }
-
-      if (serviceType.toLowerCase() === 'harmony') {
-        const {
-          hierarchyMappings,
-          keywordMappings,
-          variables
-        } = getVariables(associatedVariables)
-        const {
-          supportedOutputProjections
-        } = serviceItem
-
-        const outputFormats = []
-
-        if (supportedReformattings) {
-          supportedReformattings.forEach((reformatting) => {
-            const { supportedOutputFormats } = reformatting
-
-            // Collect all supported output formats from each mapping
-            outputFormats.push(...supportedOutputFormats)
-          })
-        }
-
-        let outputProjections = []
-        if (supportedOutputProjections) {
-          outputProjections = supportedOutputProjections.filter((projection) => {
-            const { projectionAuthority } = projection
-
-            return projectionAuthority != null
-          }).map((projection) => {
-            const { projectionAuthority } = projection
-
-            return projectionAuthority
-          })
-        }
+  const buildMethods = {
+    esi: (serviceItem, params) => buildEsiEcho(serviceItem, params.disableOrdering),
+    'echo orders': (serviceItem, params) => buildEsiEcho(serviceItem, params.disableOrdering),
+    opendap: (serviceItem, params) => buildOpendap(serviceItem, params.associatedVariables),
+    // eslint-disable-next-line max-len
+    harmony: (serviceItem, params) => buildHarmony(serviceItem, params.associatedVariables, params.harmonyIndex),
+    swodlr: (serviceItem, params) => buildSwodlr(serviceItem, params.disableSwodlr),
+    downloads: () => buildDownload(granules, isOpenSearch)
+  }
 
-        accessMethods[`harmony${harmonyIndex}`] = {
-          description,
-          enableTemporalSubsetting: true,
-          enableSpatialSubsetting: true,
-          hierarchyMappings,
-          id: serviceConceptId,
-          isValid: true,
-          keywordMappings,
-          longName,
-          name,
-          supportedOutputFormats: uniq(outputFormats),
-          supportedOutputProjections: outputProjections,
-          supportsBoundingBoxSubsetting: supportsBoundingBoxSubsetting(serviceItem),
-          supportsShapefileSubsetting: supportsShapefileSubsetting(serviceItem),
-          supportsTemporalSubsetting: supportsTemporalSubsetting(serviceItem),
-          supportsVariableSubsetting: supportsVariableSubsetting(serviceItem),
-          supportsConcatenation: supportsConcatenation(serviceItem),
-          defaultConcatenation: defaultConcatenation(serviceItem),
-          enableConcatenateDownload: defaultConcatenation(serviceItem),
-          type: serviceType,
-          url: urlValue,
-          variables
-        }
+  const nonDownloadMethods = serviceItems !== null ? serviceItems.reduce((methods, serviceItem) => {
```

Comment:
This ternary is very hard to read. I tend to avoid ternaries if they involve any kind of extra function call, `reduce` in this instance. Being a little more verbose with `if` statements can be easier to read and understand.

If you don't want to go that route, at least invert the initial condition, so that the more simple return is first, like this 
```
const nonDownloadMethods = serviceItems === null
    ? {}
    : serviceItems.reduce(.....)
```